### PR TITLE
Adding V2 Constructors in Paramaters and Emits Parameters

### PR DESCRIPTION
**Pull request by brrmorre**
**Related issue**: 

**Type of change**: bug report | feature request | other enhancement

**Impact**: no functional change | API addition (no impact on existing code) | API modification

**Development Phase**: proposal |  implementation

**Release Notes**

### CODE REVIEW

File: src/main/scala/amba/ahb/ToTL.scala [hunk 1]
```diff
@@ -11,13 +11,17 @@ import freechips.rocketchip.util._
...
+      masters = mp.masters.map { m =>
         // AHB fixed length transfer size maximum is 16384 = 1024 * 16 bits, hsize is capped at 111 = 1024 bit transfer size and hburst is capped at 111 = 16 beat burst
-        // This master can only produce:
-        // emitsGet = TransferSizes(1, 2048),
-        // emitsPutFull = TransferSizes(1, 2048)
-          TLMasterParameters.v1(name = m.name, nodePath = m.nodePath)
+          TLMasterParameters.v2(
+            name     = m.name,
+            nodePath = m.nodePath,
+            emits    = TLMasterToSlaveTransferSizes(
```
Comment (brrmorre) [inline src/main/scala/amba/ahb/ToTL.scala:20]: This node should pick a bus width that constrains this number

File: src/main/scala/diplomacy/Parameters.scala [hunk 2]
```diff
@@ -97,8 +97,9 @@ case class TransferSizes(min: Int, max: Int)
   def intersect(x: TransferSizes) =
     if (x.max < min || max < x.min) TransferSizes.none
     else TransferSizes(scala.math.max(min, x.min), scala.math.min(max, x.max))
-  
+
   // Not a union, because the result may contain sizes contained by neither term
+  // NOT TO BE CONFUSED WITH COVERPOINTS
   def cover(x: TransferSizes) = {
```
Comment (brrmorre) [inline src/main/scala/diplomacy/Parameters.scala:103]: replace this with mincover

File: src/main/scala/tilelink/Fragmenter.scala [hunk 3]
```diff
@@ -54,22 +54,24 @@ class TLFragmenter(val minSize: Int, val maxSize: Int, val alwaysMin: Boolean =
...
-        // emitsLogical = c.clients.map(_.knownToEmit.get.emitsLogical).reduce(_ smallestintervalcover _),
-        // emitsGet = c.clients.map(_.knownToEmit.get.emitsGet).reduce(_ smallestintervalcover _),
-        // emitsPutFull = c.clients.map(_.knownToEmit.get.emitsPutFull).reduce(_ smallestintervalcover _),
-        // emitsPutPartial = c.clients.map(_.knownToEmit.get.emitsPutPartial).reduce(_ smallestintervalcover _),
-        // emitsHint = c.clients.map(_.knownToEmit.get.emitsHint).reduce(_ smallestintervalcover _)
-    },
-    managerFn = { m => m.v1copy(managers = m.managers.map(mapManager)) })
+        requestFifo = true,
+        emits       = TLMasterToSlaveTransferSizes(
+          acquireT    = c.masters.map(_.emits.acquireT)  .reduce(_ cover _),
```
Comment (brrmorre) [inline src/main/scala/tilelink/Fragmenter.scala:63]: try and use shrinktransfer to make the bound tighter

Comment (terpstra) [inline src/main/scala/tilelink/Fragmenter.scala:63]: Maybe: = shrinkTransfer(c.masters.map(_.emits.acquireT)  .reduce(_ cover _)),

File: src/main/scala/tilelink/Parameters.scala [hunk 4]
```diff
@@ -386,6 +415,35 @@ object TLSlaveParameters {
       alwaysGrantsT = alwaysGrantsT,
       fifoId        = fifoId)
   }
+
+  def v2(
+    address:       Seq[AddressSet],
+    nodePath:      Seq[BaseNode]                = Seq(),
+    resources:     Seq[Resource]                = Seq(),
+    setName:       Option[String]               = None,
```
Comment (brrmorre) [inline src/main/scala/tilelink/Parameters.scala:423]: try and change setname to name

**Comment by jerryhethatday** [comment]
Can you clarify from higher level what the v2 is? why it is even needed?

**Pull request merged**